eppy3000/__init__.py

```diff
@@ -5,3 +5,47 @@
 __author__ = """Santosh Philip"""
 __email__ = "santosh@noemail.com"
 __version__ = "0.1.4"
+
+import os
+
+from .modelmaker import EPJ
+
+
+def is_path(arg):
+    try:
+        os.fspath(arg)
+        return True
+    except TypeError:
+        return False
+
+
+def get_filehandler(arg):
+    if is_path(arg):
+        return open(arg, "rt")
+    elif hasattr(arg, "read"):
+        return arg
+    else:
+        raise TypeError(
+            f"expected str, bytes, os.PathLike object or file object, not {type(arg)}"
+        )
+
+
+def epj(epj, epschema=None):
```

Comment:
Main differences from the code in the issue page is the separated consideration of epj and epschema, which makes it a bit more verbose. The current structure (with other two functions) is the most concise and clean one in my mind currently, do let me know if you any other ideas.